cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/PropertiesResolver.java

```diff
@@ -74,6 +74,7 @@ public Map<String, String> getSystemProperties(ProgramId id)
     throws IOException, NamespaceNotFoundException, AccessException {
     Map<String, String> systemArgs = new HashMap<>();
     systemArgs.put(Constants.CLUSTER_NAME, cConf.get(Constants.CLUSTER_NAME, ""));
+    systemArgs.put(ProgramOptionConstants.RUNTIME_NAMESPACE, id.getNamespace());
```

Comment:
I don't think we need to put it here, as the programId is passed all the way down as well.

Can we add it directly to the twillConfigs in the DistributedProgramRunner before calling withConfiguration()?

Reply:
Fixed